openwebstart/src/main/java/com/openwebstart/proxy/WebStartProxySelector.java

```diff
@@ -55,7 +59,15 @@ private ProxyProvider createProvider(final DeploymentConfiguration config) {
 
     @Override
     public void connectFailed(final URI uri, final SocketAddress sa, final IOException ioe) {
-        DialogFactory.showErrorDialog(Translator.getInstance().translate("proxy.error.connectionFailed", uri), ioe);
-        JNLPRuntime.exit(-1);
+            assert proxyList != null && proxyList.size() != 0;
+
+            final String currentSocketAddress = sa != null ? sa.toString() : "DIRECT";
+            final List<String> proxyAddresses = proxyList.stream().map(p -> p.address() != null ? p.address().toString() : "DIRECT").collect(Collectors.toList());
+            LOG.debug("Connection failed for proxy {} out of  {}", currentSocketAddress, proxyAddresses);
+            // if failed proxy is the last in the list means all proxies in the list have failed
+            if (proxyAddresses.get(proxyAddresses.size() - 1).equals(currentSocketAddress)) {
```

Comment:
Maybe a stupid question. But how would OWS behave if we would not call `exit(1)`?

Reply:
Not a stupid question it turns out. OWS keeps on throwing up Connection Failure dialog several times but ultimately starts the app! Does this mean that it ignores the proxy and goes DIRECT?